include/oneapi/dpl/internal/dynamic_selection_impl/fixed_resource_policy.h

```diff
@@ -0,0 +1,127 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_STATIC_POLICY_IMPL_H
+#define _ONEDPL_STATIC_POLICY_IMPL_H
+#include <vector>
+#include <type_traits>
+#include <memory>
+#include <exception>
+#include "oneapi/dpl/internal/dynamic_selection_impl/scoring_policy_defs.h"
+#if _DS_BACKEND_SYCL != 0
+    #include "oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h"
+#endif
+namespace oneapi {
+namespace dpl {
+namespace experimental {
+
+#if _DS_BACKEND_SYCL != 0
+  template <typename Backend=sycl_backend>
+#else
+  template <typename Backend>
+#endif
+  struct fixed_resource_policy {
+  private:
+    using backend_t = Backend;
+    using resource_container_t = typename backend_t::resource_container_t;
+    using execution_resource_t = typename backend_t::execution_resource_t;
+    using wrapped_resource_t = typename std::decay_t<Backend>::execution_resource_t;
+
+  public:
+    //policy traits
+    using resource_type = decltype(unwrap(std::declval<wrapped_resource_t>()));
```

Comment:
Free standing, which also means that a header include is missing. We are adding that. And with the refactoring into dynamic_selection_traits.h this will not introduce a cycle.

Reply:
Missing include added in fixed_resource_policy.h and round_robin_policy.h. We can revisit later if we want to use fully qualified names.